generate-logic-legacy/src/main/kotlin/Directories.kt

```diff
@@ -0,0 +1,8 @@
+import org.gradle.api.file.Directory
+import org.gradle.api.file.DirectoryProperty
+
+val DirectoryProperty.rootGeneratedDir: Directory
+    get() = dir("generated").get()
```

Comment:
It is a bit strange to unwrap a property to its value forcely. Most gradle configs accept both `T | Property<T>`.